packages/common/utils/src/object/pick.ts

```diff
@@ -0,0 +1,11 @@
+/** @tossdocs-ignore */
+import { ArrayUnion, ObjectKeys, objectKeys } from '.';
+
+export function pick<ObjectType extends Record<PropertyKey, unknown>, KeyTypes extends Array<ObjectKeys<ObjectType>>>(
+  obj: ObjectType,
+  keys: KeyTypes
+) {
+  return objectKeys(obj)
+    .filter((k): k is Exclude<ObjectKeys<ObjectType>, ArrayUnion<KeyTypes>> => keys.includes(k))
+    .reduce((acc, key) => ((acc[key] = obj[key]), acc), {} as Pick<ObjectType, ArrayUnion<KeyTypes>>);
+}
```

Comment:
I think it doesn't have to iterate over all keys in `obj`.

This would be more efficient:

```typescript
const picked = {};
for (const key of keys) {
  picked[key] = obj[key];
}
return picked as Pick<ObjectType, ArrayUnion<KeyTypes>>;
```

Reply:
![image](https://user-images.githubusercontent.com/69495129/202587348-ab786c4f-bf6d-491d-8547-460daf3bdf96.png)

your code is 80% faster than my code.
I learned it! thank you

Reply:
readability is also increased!

Reply:
I refactored the[`omit`](https://github.com/toss/slash/blob/main/packages/common/utils/src/object/omit.ts) function similarly, but the performance is rather poor I think it's because the object was copied.

![image](https://user-images.githubusercontent.com/69495129/202590243-2c50353d-702a-418c-ab33-bd352757a2ce.png)